src/core/achievements.cpp

```diff
@@ -1147,6 +1161,9 @@ void Achievements::ClientLoadGameCallback(int result, const char* error_message,
 
   s_state.load_game_request = nullptr;
 
+  // Move out pending progress to restore (if any) so it's cleared regardless of success/failure.
+  std::vector<u8> progress_to_restore = std::move(s_state.pending_progress_restore);
```

Comment:
_nitpick:_ the C++ standard doesn't guarantee that a vector will be empty after a move (even though that's probably the only reasonable way to implement the operation). I suggest something like this instead:

```cpp
std::vector<u8> progress_to_restore;
progress_to_restore.swap(s_state.pending_progress_restore);
```